crates/anvil/src/eth/pool/mod.rs

```diff
@@ -166,6 +166,16 @@ impl Pool {
         dropped
     }
 
+    pub fn drop_all_transactions(&self) {
+        let pool = self.inner.write();
+        let mut transactions_iterator = pool.pending_transactions.transactions();
+        
+        while let Some(pool_transaction) = transactions_iterator.next() {
+            let tx_hash = pool_transaction.as_ref().hash();
+            self.drop_transaction(tx_hash);
+        }
```

Comment:
hmm, I think we want a `fn clear` on both tx sets 

https://github.com/foundry-rs/foundry/blob/d8a162581aa2f674e581a57daf1aba24acc26206/crates/anvil/src/eth/pool/mod.rs#L202-L203

that clear all maps